File: LivrusVirtus/VCPP6/FoxToolKit/ODBC/CadAutores.cpp
#include "resource.h"
#include "Cadastros.h"
#include "CadAutores.h"
#include "Principal.h"
#include "RotinasGlobais.h"
#include "ConsultasSQL.h"

CfrmCadAutores *frmCadAutores = (CfrmCadAutores *) NULL;

FXDEFMAP(CfrmCadAutores) CfrmCadAutoresMap[]={
  FXMAPFUNC(SEL_COMMAND, CfrmCadAutores::ID_FRMCADASTROS_BTNNOVO, 
    CfrmCadAutores::onBtnNovo),
  FXMAPFUNC(SEL_COMMAND, CfrmCadAutores::ID_FRMCADASTROS_BTNPESQUISAR, 
    CfrmCadAutores::onBtnPesquisar),
  FXMAPFUNC(SEL_COMMAND, CfrmCadAutores::ID_FRMCADASTROS_BTNANTERIOR, 
    CfrmCadAutores::onBtnAnterior),
  FXMAPFUNC(SEL_COMMAND, CfrmCadAutores::ID_FRMCADASTROS_BTNPROXIMO, 
    CfrmCadAutores::onBtnProximo),
  FXMAPFUNC(SEL_COMMAND, CfrmCadAutores::ID_FRMCADASTROS_BTNSALVAR, 
    CfrmCadAutores::onBtnSalvar),
  FXMAPFUNC(SEL_COMMAND, CfrmCadAutores::ID_FRMCADASTROS_BTNEXCLUIR, 
    CfrmCadAutores::onBtnExcluir),
  FXMAPFUNC(SEL_COMMAND, CfrmCadAutores::ID_FRMCADASTROS_BTNLIMPAR, 
    CfrmCadAutores::onBtnLimpar),
  FXMAPFUNC(SEL_COMMAND, CfrmCadAutores::ID_FRMCADASTROS_BTNFECHAR, 
    CfrmCadAutores::onBtnFechar),
  FXMAPFUNC(SEL_CLOSE, 0, CfrmCadAutores::onClose)
};
FXIMPLEMENT(CfrmCadAutores,CfrmCadastros,CfrmCadAutoresMap,
  ARRAYNUMBER(CfrmCadAutoresMap))

CfrmCadAutores::CfrmCadAutores(FXMDIClient *clienteMDI, FXIcon *iconeMDI, 
  FXMDIMenu *menuMDI): CfrmCadastros(clienteMDI, STR_CADAUTOR, 
  iconeMDI, menuMDI, TEXTFIELD_INTEGER, true) {
  IniciaControles();
}

void CfrmCadAutores::IniciaControles() {
 this->labDescricao->setText("Autor: ");  
}

CfrmCadAutores::~CfrmCadAutores(){
  frmPrincipal->mnuCadAutores->enable();  
}

FXbool CfrmCadAutores::ValidaDados(FXbool bTodosDados)
{
if (Rotinas->ValidaCampo(edtCodigo->getText(), STR_CODNINF))
  	  return false;
  	if (bTodosDados) {
  	  if (Rotinas->ValidaCampo(edtDescricao->getText(), STR_AUTNINF))
  		return false;
  	}
  	return true;
}

long CfrmCadAutores::onClose(FXObject*,FXSelector,void*) {
  this->destroy();
  return true;
}

long CfrmCadAutores::onBtnNovo(FXObject*,FXSelector,void*) {
  btnNovoClick();
  NovoDado(ConsSQL->Autor('N', edtCodigo->getText(),""));
  return true;
}

long CfrmCadAutores::onBtnPesquisar(FXObject*,FXSelector,void*) {
  PesquisarDados(
    ConsSQL->Autor('S', edtCodigo->getText(),""), STR_AUTNENC);
  return true;
}

long CfrmCadAutores::onBtnAnterior(FXObject*,FXSelector,void*) {
  registroAnterior();
  return true;
}

long CfrmCadAutores::onBtnProximo(FXObject*,FXSelector,void*) {
  registroProximo();
  return true;
}

long CfrmCadAutores::onBtnSalvar(FXObject*,FXSelector,void*) {
  SalvarDados(ValidaDados(true),
    ConsSQL->Autor('S', edtCodigo->getText(),""),
    ConsSQL->Autor('U', edtCodigo->getText(), edtDescricao->getText()),
    ConsSQL->Autor('I', edtCodigo->getText(), edtDescricao->getText()),
    true);
  return true;
}

long CfrmCadAutores::onBtnExcluir(FXObject*,FXSelector,void*) {
  ExcluirDados(ValidaDados(false),
    ConsSQL->Autor('D', edtCodigo->getText(),""));
  return true;
}

long CfrmCadAutores::onBtnLimpar(FXObject*,FXSelector,void*) {
  btnLimparClick();
  return true;
}

long CfrmCadAutores::onBtnFechar(FXObject*,FXSelector,void*) {
  this->close();
  return true;
}

File: LivrusVirtus/FoxToolKit/FXPython/CadAutores.py
from FXPy.fox import *
from Cadastros import frmCadastros
from resource import Resource
from RotinasGlobais import RotinasGlobais
from ConsultasSQL import ConsultasSQL
import Principal

class frmCadAutores(frmCadastros):

  def __init__(self, clienteMDI, iconeMDI, menuMDI):  
    frmCadastros.__init__(self, clienteMDI, Resource.STR_CADAUTOR, 
      iconeMDI, menuMDI, TEXTFIELD_INTEGER, True)

    self.Rotinas = RotinasGlobais.Rotinas
    self.ConsSQL = ConsultasSQL.ConsSQL

    FXMAPFUNC(self, SEL_COMMAND, frmCadAutores.ID_FRMCADASTROS_BTNNOVO, 
      frmCadAutores.onBtnNovo)
    FXMAPFUNC(self, SEL_COMMAND, frmCadAutores.ID_FRMCADASTROS_BTNPESQUISAR, 
      frmCadAutores.onBtnPesquisar)
    FXMAPFUNC(self, SEL_COMMAND, frmCadAutores.ID_FRMCADASTROS_BTNANTERIOR, 
      frmCadAutores.onBtnAnterior)
    FXMAPFUNC(self, SEL_COMMAND, frmCadAutores.ID_FRMCADASTROS_BTNPROXIMO, 
      frmCadAutores.onBtnProximo)
    FXMAPFUNC(self, SEL_COMMAND, frmCadAutores.ID_FRMCADASTROS_BTNSALVAR, 
      frmCadAutores.onBtnSalvar)
    FXMAPFUNC(self, SEL_COMMAND, frmCadAutores.ID_FRMCADASTROS_BTNEXCLUIR, 
      frmCadAutores.onBtnExcluir)
    FXMAPFUNC(self, SEL_COMMAND, frmCadAutores.ID_FRMCADASTROS_BTNLIMPAR, 
      frmCadAutores.onBtnLimpar)
    FXMAPFUNC(self, SEL_COMMAND, frmCadAutores.ID_FRMCADASTROS_BTNFECHAR, 
      frmCadAutores.onBtnFechar)
    FXMAPFUNC(self, SEL_CLOSE, 0, frmCadAutores.onClose)

    self.labDescricao.setText("Autor: ")

  def ValidaDados(self, bTodosDados):
    if self.Rotinas.ValidaCampo(self.edtCodigo.getText(), Resource.STR_CODNINF):
   	  return False
    if bTodosDados:
  	  if self.Rotinas.ValidaCampo(self.edtDescricao.getText(), Resource.STR_AUTNINF):
    		return False        	
    return True
  
  def onBtnNovo(self, sender, sel, ptr):
    self.btnNovoClick()
    self.NovoDado(self.ConsSQL.Autor('N', self.edtCodigo.getText(),""))	
    return True  
  
  def onBtnPesquisar(self, sender, sel, ptr):
    self.PesquisarDados(
      self.ConsSQL.Autor('S', self.edtCodigo.getText(),""), Resource.STR_AUTNENC)
    return True  
  
  def onBtnAnterior(self, sender, sel, ptr):
    self.registroAnterior()	
    return True  
  
  def onBtnProximo(self, sender, sel, ptr):
    self.registroProximo()
    return True  
  
  def onBtnSalvar(self, sender, sel, ptr):
    self.SalvarDados(self.ValidaDados(True),
      self.ConsSQL.Autor('S', self.edtCodigo.getText(),""),
      self.ConsSQL.Autor('U', self.edtCodigo.getText(), 
        self.edtDescricao.getText()),
      self.ConsSQL.Autor('I', self.edtCodigo.getText(), 
        self.edtDescricao.getText()), 
      True)
    return True  
  
  def onBtnExcluir(self, sender, sel, ptr):
    self.ExcluirDados(self.ValidaDados(False),
      self.ConsSQL.Autor('D', self.edtCodigo.getText(),""))
    return True  
  
  def onBtnLimpar(self, sender, sel, ptr):
    self.btnLimparClick()
    return True  

  def onBtnFechar(self, sender, sel, ptr):
    self.onClose(sender, sel, ptr)
    return True
  
  def onClose(self, sender, sel, ptr):
    Principal.fPrincipal.habilitaMenu("CadAutores")
    self.destroy()
    return True
